src/frontend/fit/standard_errors/hessian.jl

```diff
@@ -1,58 +1,54 @@
 """
-    se_hessian(semfit::SemFit; hessian = :finitediff)
+    se_hessian(fit::SemFit; method = :finitediff)
 
-Return hessian based standard errors.
+Return hessian-based standard errors.
 
 # Arguments
-- `hessian`: how to compute the hessian. Options are
+- `method`: how to compute the hessian. Options are
     - `:analytic`: (only if an analytic hessian for the model can be computed)
     - `:finitediff`: for finite difference approximation
 """
-function se_hessian(sem_fit::SemFit; hessian = :finitediff)
-    c = H_scaling(sem_fit.model)
-
-    if hessian == :analytic
-        par = solution(sem_fit)
-        H = zeros(eltype(par), length(par), length(par))
-        hessian!(H, sem_fit.model, sem_fit.solution)
-    elseif hessian == :finitediff
-        H = FiniteDiff.finite_difference_hessian(
-            Base.Fix1(objective!, sem_fit.model),
-            sem_fit.solution,
-        )
-    elseif hessian == :optimizer
-        throw(
-            ArgumentError(
-                "standard errors from the optimizer hessian are not implemented yet",
-            ),
-        )
-    elseif hessian == :expected
-        throw(
-            ArgumentError(
-                "standard errors based on the expected hessian are not implemented yet",
-            ),
+function se_hessian(fit::SemFit; method = :finitediff)
+    c = H_scaling(fit.model)
+    params = solution(fit)
+    H = similar(params, (length(params), length(params)))
+
+    if method == :analytic
+        evaluate!(nothing, nothing, H, fit.model, params)
+    elseif method == :finitediff
+        FiniteDiff.finite_difference_hessian!(
+            H,
+            p -> evaluate!(zero(eltype(H)), nothing, nothing, fit.model, p),
+            params,
         )
+    elseif method == :optimizer
+        error("Standard errors from the optimizer hessian are not implemented yet")
+    elseif method == :expected
+        error("Standard errors based on the expected hessian are not implemented yet")
     else
-        throw(ArgumentError("I don't know how to compute `$hessian` standard-errors"))
+        throw(ArgumentError("Unsupported hessian calculation method :$method"))
     end
 
-    invH = c * inv(H)
-    se = sqrt.(diag(invH))
-
-    return se
+    H_chol = cholesky!(Symmetric(H))
+    H_inv = LinearAlgebra.inv!(H_chol)
+    return [sqrt(c * H_inv[i]) for i in diagind(H_inv)]
```

Comment:
I don't know why, but this seems to be ~10 times faster and requires less memory^^:
```suggestion
    return sqrt.(c * diag(H_inv))
```

Reply:
Interesting. Probably has something to do with the bound checks, but I'll try to clarify whether it's so slow because of the type inference problems.
The whole idea was to save some array allocations :)

Reply:
Hmm, actually, in the benchmark the version from the PR has less allocations:

```
function test_se_hessian(H)
    c = 1.5
    H_chol = cholesky!(Symmetric(H))
    H_inv = LinearAlgebra.inv!(H_chol)
    return [sqrt(c*H_inv[i]) for i in diagind(H_inv)]
end

function test_se_hessian2(H)
    c = 1.5
    H_chol = cholesky!(Symmetric(H))
    H_inv = LinearAlgebra.inv!(H_chol)
    return sqrt.(c .* diag(H_inv))
end

using LinearAlgebra
using BenchmarkTools

n = 100
H = randn(n, n)
H = H * H'

```
@benchmark test_se_hessian($H)

BenchmarkTools.Trial: 10000 samples with 1 evaluation.
 Range (min … max):  264.733 μs …  2.782 ms  ┊ GC (min … max): 0.00% … 0.00%
 Time  (median):     312.433 μs              ┊ GC (median):    0.00%
 Time  (mean ± σ):   329.965 μs ± 71.296 μs  ┊ GC (mean ± σ):  0.00% ± 0.00%

   ▂█▂                                                          
  ▂████▇▅▄▃▃▃▄▄▅▅▅▄▄▃▃▃▃▃▂▃▃▃▂▂▃▂▂▂▂▂▂▂▂▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁ ▂
  265 μs          Histogram: frequency by time          545 μs <

 Memory estimate: 896 bytes, allocs estimate: 1.

@benchmark test_se_hessian2($H)

BenchmarkTools.Trial: 10000 samples with 1 evaluation.
 Range (min … max):  264.373 μs …  3.936 ms  ┊ GC (min … max): 0.00% … 0.00%
 Time  (median):     327.932 μs              ┊ GC (median):    0.00%
 Time  (mean ± σ):   341.987 μs ± 90.513 μs  ┊ GC (mean ± σ):  0.00% ± 0.00%

    ▇█▂▁  ▅▃▂▃▁    ▁▁                                           
  ▁▄████▆▇█████▇▇▇████▇▇▆▅▅▄▄▄▄▃▃▃▃▂▂▂▂▂▂▂▂▂▂▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁ ▃
  264 μs          Histogram: frequency by time          552 μs <

 Memory estimate: 1.75 KiB, allocs estimate: 2.
```

```

Reply:
Sorry, this was my mistake - I accidentally benchmarked in a global scope, and this makes a huge difference for `[sqrt(c*H_inv[i]) for i in diagind(H_inv)]`.